src/java.base/share/classes/java/util/stream/StreamSupport.java

```diff
@@ -315,4 +316,49 @@ public static DoubleStream doubleStream(Supplier<? extends Spliterator.OfDouble>
                                          StreamOpFlag.fromCharacteristics(characteristics),
                                          parallel);
     }
+
+    /**
+     * Creates a new sequential empty {@code Stream} with the
+     * characteristics of the provided {@code Spliterator}.
+     *
+     * @param <T> the type of stream elements
+     * @param spliterator a {@code Spliterator} describing the stream elements
+     * @return a new sequential empty {@code Stream}
+     */
+    public static <T> Stream<T> emptyStream(Spliterator<T> spliterator) {
+        return new Streams.EmptyStream<>(spliterator);
+    }
+
+    /**
+     * Creates a new sequential empty {@code IntStream} with the
+     * characteristics of the provided {@code Spliterator}.
+     *
+     * @param spliterator a {@code Spliterator.OfInt} describing the stream elements
+     * @return a new sequential empty {@code IntStream}
+     */
+    public static IntStream emptyIntStream(Spliterator.OfInt spliterator) {
```

Comment:
In the interest of not adding new public APIs, can these be removed in favour of eg, IntStream.empty()? It wouldn’t take the spliterator, but perhaps it isn’t necessary.